src/camera.py

```diff
@@ -0,0 +1,212 @@
+"""
+camera
+
+Author: Steven Dropiewski
+"""
+from os import path
+import sys
+import time
+
+FILE_PATH = path.dirname(path.realpath(__file__))
+SUNFOUNDER_PATH = "SunFounder_PiCar-V/remote_control/remote_control/driver"
+sys.path.append(FILE_PATH + "/../" + SUNFOUNDER_PATH)
+
+from picar.SunFounder_PCA9685 import Servo
+from picar import filedb
+import picar
+
+class Camera:
+    """
+    Camera class to control panning and tilting.
+
+    Creates camera object that can be operated on using set_pan and set_tilt.
+    set_pan operates [0,180] degrees. set_tilt operates [0,150] degrees. The 
+    current values for pan and tilt in degrees can be returned using the get_pan
+    and get_tilt methods. To change the offset you can may use calibrate_pan
+    and calibrate_tilt methods. The methods will automatically update the values
+    in the config file. All method parameters take absolute degrees.
+    
+    :param bus_number: The bus number the hat connects to.
+    :type bus_number: int
+    """
+
+    PAN_CHANNEL = 1
+    """Channel on PiCar HATS that connects to pan servo."""
+    TILT_CHANNEL = 2
+    """Channel on PiCar HATS that connects to tilt servo."""
+    MAX_PAN_DEGREE = 180
+    """The maximum value to be written to pan servo (in degrees)."""
+    MAX_TILT_DEGREE = 150
+    """The maximum value to be written to tilt servo (in degrees)."""
+    MIN_DEGREE = 0
+    """The minimum value to be written to both pan and tilt servo (in degrees)."""
+    MAX_CALI_DEGREE = 180
+    """The maximum value to calibrate pan and tilt (in degrees)."""
+    MIN_CALI_DEGREE = -180
+    """The minimum value to calibrate pan or tilt (in degrees."""
+    OFFSET_TO_SERVO = 0
+    """The offset value passed to the servo object."""
+
+    def __init__(self, bus_number=1):
+        picar.setup()
+
+        self._DB = filedb.fileDB(FILE_PATH + "/../" + SUNFOUNDER_PATH + "/config")
+
+        self._pan_offset = float(self._DB.get('pan_offset', default_value=0))
+        self._tilt_offset = float(self._DB.get('tilt_offset', default_value=0))
+
+        self._pan_servo = Servo.Servo(self.PAN_CHANNEL,  
+                                      bus_number=bus_number, 
+                                      offset=self.OFFSET_TO_SERVO)
+        self._tilt_servo = Servo.Servo(self.TILT_CHANNEL, 
+                                       bus_number=bus_number, 
+                                       offset=self.OFFSET_TO_SERVO)
+
+        self._current_pan_angle = 0
+        self._current_tilt_angle = 0
+
+
+    def set_pan(self, degree):
+        """
+        Pans the camera.
+
+        Controls the horizontal movement of the camera in [0, 180] degrees. 
+        0 refers to the vehicle's right while 180 refers to the vehicle's left. 
+
+        :param degree: The degree to set the pan servo to.
+        :type degree: float
```

Comment:
I assume via computer vision we will be getting a lot of no integer values. I tested it with floats and it does use them and works correctly.

Reply:
yeah but I'm pretty sure all it does is truncate the float part (the hardware isn't that good). It's expected that they'll use ints as the hardware doesn't support panning to say angle 1.2345. It can be used that way, but we shouldn't say that in our documentation.